### DISCUSSION TITLE
Declaring form fields using Pydantic models does not work with `TypeAliasType` in python 3.12

### DISCUSSION THREAD

**Discussion by 9en9i** [fastapi/fastapi#12149] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import TypeVar, Annotated

from fastapi import FastAPI, Form, UploadFile
from pydantic import BaseModel

app = FastAPI()

class NewForm(BaseModel):
    field: str | None = None

type InjectForm[T] = Annotated[T, Form(media_type="multipart/form-data")]

@app.post("/api")
def func(form: InjectForm[NewForm]):
    return form.model_dump(exclude_unset=True)
```

### Description

Hi, first of all, thanks for the new release! I found a small bug (you could say cosmetic, I guess).
The new function from release 0.113.0 to declare form fields using Pydantic does not work with `TypeAliasType` syntax. Problems in both swagger and api client (postman for example).

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/46961e2a-fab3-40aa-8829-b84b09023f19">

<img width="1441" alt="image" src="https://github.com/user-attachments/assets/df76e7b8-305f-48ae-b175-223648c4101d">

### Operating System

macOS

### FastAPI Version

0.113.0

### Pydantic Version

2.9.0

### Python Version

Python 3.12.2

**Comment by YuriiMotov** [comment]
Yes, it's not fully supported yet.
For now use old syntax:
```
T = TypeVar("T", bound=BaseModel)
InjectForm = Annotated[T, Form(media_type="multipart/form-data")]

```

**Issue to track this:** https://github.com/fastapi/fastapi/issues/10719
